### PR TITLE
support tfdataset without batch

### PR DESCRIPTION
**Pull request by yangw1234**
support tfdataset without batch:
if user does not specify batch_size in TFDataset, then TFDataset.tensors will return the a tensor without batch dim.

### CODE REVIEW

File: pyzoo/zoo/tfpark/estimator.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def predict(self, input_fn, checkpoint_path=None):
                         saver.restore(sess, checkpoint_path)
                     else:
                         sess.run(tf.global_variables_initializer())
-                    inputs = nest.flatten(result.feature_tensors)
+                    inputs = nest.flatten(result._original_tensors[0])
```
Comment (jason-dai) [inline pyzoo/zoo/tfpark/estimator.py]: I think we should require the dataset has batch when used for training

File: pyzoo/zoo/tfpark/estimator.py [hunk 2]
```diff
@@ -168,7 +168,7 @@ def evaluate(self, input_fn, eval_methods, steps=None, checkpoint_path=None):
...
```
Comment (jason-dai) [inline pyzoo/zoo/tfpark/estimator.py]: Do we allow `evaluate` without batch? Do we have unit test for it?

**Comment by jason-dai** [comment]
I think we should require the dataset has batch when used for training (and maybe also for KerasModel)?

File: pyzoo/zoo/tfpark/estimator.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/tfpark/estimator.py]: Being without batch is disallowed in KerasModel and estimator training. Estimator evaluate and predict is allowed. Unit tests are added.

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2109/
